src/algoliasearch.helper.js

```diff
@@ -262,14 +262,16 @@ AlgoliaSearchHelper.prototype.searchOnce = function(options, cb) {
  * See the description of [FacetSearchResult](reference.html#FacetSearchResult)
  * @param {string} facet the name of the faceted attribute
  * @param {string} query the string query for the search
- * @param {number} maxFacetHits the maximum number values returned. Should be > 0 and <= 100
- * @return {promise<FacetSearchResult>} the results of the search
- */
-AlgoliaSearchHelper.prototype.searchForFacetValues = function(facet, query, maxFacetHits) {
-  var state = this.state;
-  var index = this.client.initIndex(this.state.index);
+ * @param {number} [maxFacetHits] the maximum number values returned. Should be > 0 and <= 100
+ * @param {object} [userState] the set of custom parameters to use on top of the current state. Setting a property to `undefined` removes
+ * it in the generated query.
```

Comment:
We should maybe accept `null` value as well?

Reply:
Oh reading next line:

```js
var state = this.state.setQueryParameters(userState || {});
```

We can maybe update the documentation to:

> Setting a falsy property (`null`, `false`, `undefined`, `''`) removes it in the generated query

Reply:
I don't think we should care about the variance here. Let's keep it `undefined` IMO

Reply:
Ok so interestingly, `undefined` is the only falsy value that let you unset a state property.